src/sidebar/components/test/tag-editor-test.js

```diff
@@ -1,63 +1,257 @@
 'use strict';
 
-const angular = require('angular');
+const { createElement } = require('preact');
+const { mount } = require('enzyme');
 
-const util = require('../../directive/test/util');
+const mockImportedComponents = require('./mock-imported-components');
+const TagEditor = require('../tag-editor');
 
-describe('tagEditor', function() {
-  let fakeTags;
+describe('TagEditor', function() {
+  let fakeTags = ['tag1', 'tag2'];
+  let fakeTagsService;
+  let fakeServiceUrl;
+  let fakeOnEditTags;
 
-  before(function() {
-    angular.module('app', []).component('tagEditor', require('../tag-editor'));
-  });
+  function createComponent(props) {
+    return mount(
+      <TagEditor
+        // props
+        annotation={{
+          id: 'annotation_id',
+        }}
+        onEditTags={fakeOnEditTags}
+        tagList={fakeTags}
+        // service props
+        serviceUrl={fakeServiceUrl}
+        tags={fakeTagsService}
+        {...props}
+      />
+    );
+  }
 
   beforeEach(function() {
-    fakeTags = {
-      filter: sinon.stub(),
+    fakeOnEditTags = sinon.stub();
+    fakeServiceUrl = sinon.stub().returns('http://serviceurl.com');
+    fakeTagsService = {
+      filter: sinon.stub().returns(['tag3', 'tag4']),
       store: sinon.stub(),
     };
 
-    angular.mock.module('app', {
-      tags: fakeTags,
+    TagEditor.$imports.$mock(mockImportedComponents());
+  });
+
+  afterEach(() => {
+    TagEditor.$imports.$restore();
+  });
+
+  it('adds appropriate tag values to the elements', () => {
+    const wrapper = createComponent();
+    wrapper.find('li').forEach((tag, i) => {
+      assert.isTrue(tag.hasClass('tag-editor__tag-item'));
+      assert.equal(tag.text(), fakeTags[i]);
+      assert.equal(tag.prop('aria-label'), `Tag: ${fakeTags[i]}`);
     });
   });
 
-  it('converts tags to the form expected by ng-tags-input', function() {
-    const element = util.createDirective(document, 'tag-editor', {
-      tags: ['foo', 'bar'],
+  it('creates the component with a unique id', () => {
+    const wrapper = createComponent();
+    assert.equal(
+      wrapper.find('input').prop('list'),
+      'tag-editor-datalist-annotation_id'
+    );
+  });
+
+  it('calls fakeTagsService.filter with the value from the input field', () => {
+    const wrapper = createComponent();
+    wrapper.find('input').instance().value = 'tag3';
+    // simulate `keyup` to populate suggestions list
+    wrapper.find('input').simulate('keyup', { key: 'any' });
+    assert.isTrue(fakeTagsService.filter.calledWith('tag3'));
+  });
+
+  it('generates a datalist set equal to the array value returned from fakeTagsService.filter ', () => {
```

Comment:
This test tests for the presence of two elements, but does not test that there are no additional values.